File: src/net/Wexflow.Core/LaunchType.cs
namespace Wexflow.Core
{
    /// <summary>
    /// Launch type.
    /// </summary>
    public enum LaunchType
    {
        /// <summary>
        /// The workflow starts when Wexflow engine starts.
        /// </summary>
        Startup,
        /// <summary>
        /// The workflow must be triggered manually to start.
        /// </summary>
        Trigger,
        /// <summary>
        /// The workflow starts periodically.
        /// </summary>
        Periodic,
        /// <summary>
        /// The workflow starts depending on the cron scheduling expression
        /// </summary>
        Cron
    }
}

File: src/android/app/src/main/java/com/wexflow/LaunchType.java
package com.wexflow;

enum LaunchType {
    Startup,
    Trigger,
    Periodic,
    Cron;

    public static LaunchType fromInteger(int x) {
        switch (x) {
            case 0:
                return Startup;
            case 1:
                return Trigger;
            case 2:
                return Periodic;
            case 3:
                return Cron;
            default:
                return null;
        }
    }
}
